### PR TITLE
app: os-updates

### PR DESCRIPTION
**Pull request by crcro**
initial os-updates app; includes support for ubuntu/centos/opensuse

### CODE REVIEW

**Comment by paulgear** [comment]
Hi @crcro, please use data_update() instead of calling influx_update() and rrdtool_update() separately.  Please also use rrd_name() instead of creating the RRD filename manually.

File: includes/polling/applications/os-updates.inc.php [hunk 1]
```diff
@@ -21,6 +21,6 @@
     'packages' => $osupdates,
 );
 
-rrdtool_update($rrd_filename, $fields);
 $tags = array('name' => 'os-updates', 'app_id' => $app['app_id']);
-influx_update($device,'app',$tags,$fields);
+
+data_update($device,$filename,$tags,$fields);
```
Comment (murrant) [inline includes/polling/applications/os-updates.inc.php]: The second argument should be 'app'.

File: includes/polling/applications/os-updates.inc.php [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+$filename = array('app', 'os-updates', $app['app_id']);
+$rrd      = rrd_name($device['hostname'], $filename);
+$options  = '-O qv';
+$mib      = 'NET-SNMP-EXTEND-MIB';
+$oid	  = '.1.3.6.1.4.1.8072.1.3.2.4.1.2.8.111.115.117.112.100.97.116.101.1';
 
 $osupdates = snmp_get($device, $oid, $options, $mib);
 
-if(!is_file($rrd_filename))
+if(!is_file($rrd))
```
Comment (murrant) [inline includes/polling/applications/os-updates.inc.php]: instead of rrdtool_create, define the tag names 'rrd_def' and 'rrd_name'

**Comment by murrant** [comment]
@crcro Please see, PR #3800 for examples.

Here is apache for example specifically: https://github.com/murrant/librenms/blob/c43f533b55fd662d794234b686b569f487896078/includes/polling/applications/apache.inc.php

**Comment by crcro** [comment]
@murrant ... is there any requirement that needs to be fulfilled? (maybe rrdtool > 1.5.0?) because i changed the graphs as per your request and they won't work anymore:

also there is no variable $app['app_instance'] and the vardump for $rrd_name = array('app', $name, $app_instance) prints:

array(3) {
  [0]=>
  string(3) "app"
  [1]=>
  string(10) "os-updates"
  [2]=>
  string(0) ""
}

poller.php debug: http://pastebin.com/RSBX0ew5

**Comment by murrant** [comment]
app_instance is just set by drbd, sorry about that I tried to find a good example.

You should use `$rrd_name = array('app', $name, $app_id);`

**Comment by crcro** [comment]
@murrant fixed some things along but still complaining about an rrdtool_check_rrd_exists function not existing (also could not find a definition of this function in any file). what I am missing?

**Comment by murrant** [comment]
Thanks for making these changes.  Just use is_file for now.  rrdtool_check_rrd_exists isn't merged yet.

**Comment by crcro** [comment]
new, clean PR

**Pull request closed without merging**